ports/psoc6/machine_rtc.c

```diff
@@ -345,26 +353,26 @@ static mp_obj_t machine_rtc_irq(size_t n_args, const mp_obj_t *pos_args, mp_map_
     enum { ARG_trigger, ARG_handler, ARG_wake};
     static const mp_arg_t allowed_args[] = {
         { MP_QSTR_trigger, MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = CYHAL_RTC_ALARM} },
-        { MP_QSTR_handler, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_rom_obj = MP_ROM_NONE} },
+        { MP_QSTR_handler, MP_ARG_REQUIRED | MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_rom_obj = MP_ROM_NONE} },
         { MP_QSTR_wake, MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 0} }
     };
 
     mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
     mp_arg_parse_all(n_args - 1, pos_args + 1, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
     machine_rtc_obj_t *self = pos_args[0];
     self->callback = args[ARG_handler].u_obj;
+    // TODO: Shall we do some argument validation here?
```

Comment:
Ideally callback can be NULL or a pointer to handler function. For ex: A NULL is valid, when wake-up condition is set. You only want to wake up the system from deep-sleep/hibernate modes and set the wakeup source as RTC ALARM and then you do not need user-level cback. The only difference is if handler is NULL, we don't need to call register function. Rest remains the same.